internal/provider/config_data_source.go

```diff
@@ -0,0 +1,105 @@
+package provider
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+
+	apkotypes "chainguard.dev/apko/pkg/build/types"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"gopkg.in/yaml.v2"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &ConfigDataSource{}
+
+func NewConfigDataSource() datasource.DataSource {
+	return &ConfigDataSource{}
+}
+
+// ConfigDataSource defines the data source implementation.
+type ConfigDataSource struct {
+}
+
+// ConfigDataSourceModel describes the data source data model.
+type ConfigDataSourceModel struct {
+	Id     types.String        `tfsdk:"id"`
+	Config types.String        `tfsdk:"config"`
+	Data   *ImageConfiguration `tfsdk:"data"`
+}
+
+type ImageConfiguration struct {
+	// TODO(mattmoor): Add the rest of the fields
+	// from types.ImageConfiguration
+	Archs []types.String `tfsdk:"archs"`
+}
+
+func (d *ConfigDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_config"
+}
+
+func (d *ConfigDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This reads an apko configuration file into a structured form.",
+		Attributes: map[string]schema.Attribute{
+			"config": schema.StringAttribute{
+				MarkdownDescription: "The apko configuration file.",
+				Required:            true,
+			},
```

Comment:
Maybe `config_file`? `config_contents`?

So that `data` can become `config` or `parsed_config` or something more descriptive?

Reply:
I'm going with `config_contents` in and `config` out